api/v0/finder/client/http/client.go

```diff
@@ -37,37 +41,161 @@ func New(baseURL string, options ...httpclient.Option) (*Client, error) {
+	if err != nil {
+		return nil, err
+	}
+	_ = resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, httpclient.ReadError(resp.StatusCode, body)
+	}
+
+	privateFindResponse := &model.PrivateFindResponse{}
+	err = json.Unmarshal(body, privateFindResponse)
+	if err != nil {
+		return nil, err
+	}
 
-	return c.sendRequest(req)
+	return c.buildFindResponse(ctx, []multihash.Multihash{mh}, privateFindResponse)
 }
 
 // FindBatch queries indexer entries for a batch of multihashes
 func (c *Client) FindBatch(ctx context.Context, mhs []multihash.Multihash) (*model.FindResponse, error) {
 	if len(mhs) == 0 {
 		return &model.FindResponse{}, nil
 	}
-	data, err := model.MarshalFindRequest(&model.FindRequest{Multihashes: mhs})
+
+	smhs := make([]multihash.Multihash, 0, len(mhs))
+	for _, mh := range mhs {
+		smh, err := dhash.SecondMultihash(mh)
+		if err != nil {
+			return nil, err
+		}
+		smhs = append(smhs, smh)
+	}
+
+	reqBody, err := model.MarshalFindRequest(&model.FindRequest{Multihashes: smhs})
 	if err != nil {
 		return nil, err
 	}
-	req, err := http.NewRequestWithContext(ctx, http.MethodPost, c.finderURL, bytes.NewBuffer(data))
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, c.finderURL, bytes.NewBuffer(reqBody))
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Add("Accept", "application/json")
+
+	resp, err := c.c.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	resBody, err := io.ReadAll(resp.Body)
 	if err != nil {
 		return nil, err
 	}
-	return c.sendRequest(req)
+	_ = resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, httpclient.ReadError(resp.StatusCode, resBody)
+	}
+
+	privateFindResponse := &model.PrivateFindResponse{}
+	err = json.Unmarshal(resBody, privateFindResponse)
+	if err != nil {
+		return nil, err
+	}
+
+	return c.buildFindResponse(ctx, mhs, privateFindResponse)
+}
+
+func (c *Client) buildFindResponse(ctx context.Context, mhs []multihash.Multihash, pfr *model.PrivateFindResponse) (*model.FindResponse, error) {
+	// decrypt each value key using the original multihash
+	// then for each decrypted value key fetch provider's addr info
+	findResponse := &model.FindResponse{
+		MultihashResults: make([]model.MultihashResult, 0, len(pfr.MultihashResults)),
+	}
+	mhi := 0
+	for _, pmhr := range pfr.MultihashResults {
+		var mh multihash.Multihash
+		for {
+			if mhi >= len(mhs) {
+				break
+			}
+			mh = mhs[mhi]
+			mhi++
+			smh, err := dhash.SecondMultihash(mh)
+			if err != nil {
+				return nil, err
+			}
+			if bytes.Equal(smh, pmhr.Multihash) {
+				break
+			}
+		}
+
+		if mh == nil {
+			break
+		}
+
+		mhr := model.MultihashResult{
+			Multihash: mh,
+		}
+		for _, evk := range pmhr.ValueKeys {
+			evkBytes, err := b58.Decode(evk)
+			if err != nil {
+				return nil, err
+			}
+			vk, err := dhash.DecryptValueKey(evkBytes, mh)
+			if err != nil {
+				return nil, err
+			}
+
+			pid, _ := c.keyer.SplitKey(vk)
+
+			pinfo, err := c.GetProvider(ctx, pid)
```

Comment:
This should probably use a different API, since this was intended to get general information about a provider, and not specifically what is needed for a find query response.